plinth/modules/users/views.py

```diff
@@ -168,22 +167,25 @@ def form_valid(self, form):
         return super(UserChangePassword, self).form_valid(form)
 
 
-class State1View(CV):
+class FirstBootView(django.views.generic.CreateView):
     """Create user account and log the user in."""
-    template_name = 'firstboot_state1.html'
-    form_class = State1Form
-    success_url = ''
-
-    def __init__(self, *args, **kwargs):
-        """Initialize the view object."""
-        if not cfg.danube_edition:
-            mark_step_done('pagekite_firstboot')
-        mark_step_done('users_firstboot')
-        self.success_url = next_step()
-        return super(State1View, self).__init__(*args, **kwargs)
+    template_name = 'users_firstboot.html'
+
+    form_class = FirstBootForm
 
     def get_form_kwargs(self):
         """Make request available to the form (to insert messages)"""
-        kwargs = super(State1View, self).get_form_kwargs()
+        kwargs = super(FirstBootView, self).get_form_kwargs()
         kwargs['request'] = self.request
         return kwargs
+
+    def form_valid(self, form):
+        """Mark this first boot step as completed and save form."""
+        if User.objects.all():
+            first_boot.mark_step_done('users_firstboot')
```

Comment:
It looks like form_valid is called before the user is created. We can move this line into FirstBootForm.save().